logstash-core/lib/logstash/api/lib/app/commands/stats/hotthreads_command.rb

```diff
@@ -0,0 +1,96 @@
+# encoding: utf-8
+require "app/command"
+require 'monitoring'
+require "socket"
+
+class LogStash::Api::HotThreadsCommand < LogStash::Api::Command
+
+  def run(options={})
+    hash      = JRMonitor.threads.generate
+    ThreadDump.new(hash, options)
+  end
+
+  private
+
+  class ThreadDump
+
+    SKIPPED_THREADS = [ "Finalizer", "Reference Handler", "Signal Dispatcher" ].freeze
+
+    attr_reader :top_count, :ignore, :dump
+
+    def initialize(dump, options={})
+      @dump    = dump
+      @options = options
+      @top_count = options.fetch(:threads, 3)
+      @ignore    = options.fetch(:ignore_idle_threads, true)
+    end
```

Comment:
No sure about align of `=`,  I personally find them to be hard to maintain with changes.